packages/flutter_driver/lib/src/driver/timeline_summary.dart

```diff
@@ -145,10 +147,14 @@ class TimelineSummary {
       : json.encode(jsonObject);
   }
 
-  List<TimelineEvent> _extractNamedEvents(String name) {
-    return _timeline.events
-      .where((TimelineEvent event) => event.name == name)
-      .toList();
+  List<TimelineEvent> _extractNamedEvents(String name, {_TimelineEventFilter filter}) {
+    final Iterable<TimelineEvent> namedEvents = _timeline.events
+        .where((TimelineEvent event) => event.name == name);
+    if (filter == null) {
+      return namedEvents.toList();
+    } else {
+      return namedEvents.where(filter).toList();
+    }
```

Comment:
If you are just going to post filter the results, then this can be done outside this function to save changing this code.

If we are going to add the concept of a filter, then rather than grafting it on to this function as an add-on step, create a new extractFilteredEvents() method which takes a filter and have this call it with a "name" filter. Then the scene display lag code can use its own filter that looks at both name and the phase of the event to do its work.

This method would become `_extractFilteredEvents((TimelineEvent event) => event.name == name)`

The method down below would become `_extractFilteredEvents((TimelineEvent event) => event.name == kSceneDisplayLagEvent && event.phase == 'B')`